video/out/opengl/context_drm_egl.c

```diff
@@ -239,12 +242,22 @@ static bool init_gbm(struct ra_ctx *ctx)
 
     MP_VERBOSE(ctx->vo, "Initializing GBM surface (%d x %d)\n",
         p->draw_surface_size.width, p->draw_surface_size.height);
-    p->gbm.surface = gbm_surface_create(
-        p->gbm.device,
-        p->draw_surface_size.width,
-        p->draw_surface_size.height,
-        p->gbm_format,
-        GBM_BO_USE_SCANOUT | GBM_BO_USE_RENDERING);
+    if (p->num_gbm_modifiers == 0) {
+        p->gbm.surface = gbm_surface_create(
+            p->gbm.device,
+            p->draw_surface_size.width,
+            p->draw_surface_size.height,
+            p->gbm_format,
+            GBM_BO_USE_SCANOUT | GBM_BO_USE_RENDERING);
+    } else {
+        p->gbm.surface = gbm_surface_create_with_modifiers(
```

Comment:
I was wondering if we can rely on create_with_modifier to exist if `drmModeAddFB2WithModifiers` is assured to exist (already covered by current check).

Reply:
They are different APIs provided by separate libraries, so that's a no.

The drm API mentioned is introduced with 2.4.71 - back in 2016 - and we already require newer version.

Reply:
I've added a version check. I'm going to claim that it's reasonable to just require >= 17.1.0 than to add an #ifdef.